packages/workbox-routing/lib/Route.mjs

```diff
@@ -0,0 +1,170 @@
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import core from 'workbox-core';
+
+import {defaultMethod, validMethods} from './constants.mjs';
+
+/**
+ * This is the definition of the `match` callback passed into the
+ * `Route` constructor.
+ *
+ * This callback is used to determine if a new `fetch` event can be served
+ * by this `Route`. Returning a truthy value indicates that this `Route` can
+ * handle this `fetch` event. Return `null` if this shouldn't match against
+ * the `fetch` event.
+ *
+ * If you do return a truthy value, the object will be passed to the
+ * Route's `handler` (see the
+ * [Route Constructor]{@link module:workbox-routing.Route}).
+ *
+ * @callback Route~matchCallback
+ * @param {Object} input
+ * @param {URL} input.url The request's URL.
+ * @param {FetchEvent} input.event The event that triggered the `fetch` handler.
+ * @return {Object|null} To signify a match, return a truthy value, otherwise
+ * return null if the route shouldn't match. If you return an Object with
+ * contents it will be passed to the `handler` in the `Route` constructor.
+ * @memberof module:workbox-routing
+ */
+
+/**
+ * This is the definition of the `handler` callback that can be passed into the
+ * `Route` constructor.
+ *
+ * The `handler` callback is called when a request has been matched by
+ * a `Route` and should return a Promise that resolves with a `Response`.
+ *
+ * @callback Route~handlerCallback
+ * @param {Object} input
+ * @param {URL} input.url The request's URL.
+ * @param {FetchEvent} input.event The event that triggered the `fetch` handler.
+ * @param {Object} [input.params] Parameters returned
+ * the Route's [match callback]{@link
+  *   module:workbox-routing.Route~matchCallback} function. This will be
+ * undefined if nothing was returned.
+ * @return {Promise<Response>} The response that will fulfill the request.
+ * @memberof module:workbox-routing
+ */
+
+/**
+ * A `Route` allows you to tell a service worker that it should handle
+ * certain network requests using a specific response strategy.
+ *
+ * A consists or a matcher and a handler. A matcher needs to determine if a
+ * route should be used for a request. A handler should handle the request
+ * if it does match a Router.
+ *
+ * Instead of implementing your own handlers, you can use one of the
+ * pre-defined runtime caching strategies from the
+ * {@link module:workbox-runtime-caching|workbox-runtime-caching} module.
+ *
+ * There are also pre-defined Route's provided by this library:
+ * {@link module:workbox-routing.RegExpRoute|RegExpRoute}
+ * and {@link module:workbox-routing.ExpressRoute|ExpressRoute} subclasses
+ * which provide a convenient wrapper with a nicer interface for using regular
+ * expressions or Express-style routes as the `match` criteria.
+ *
+ * @example
```

Comment:
Given we are moving to DevSite. I'm wondering if it still makes sense to include code samples in JSDocs vs just moving to actual written docs for code samples and keep this just for reference.

Reply:
I can drop the code samples. It is one more thing to keep up to date and that might drift as the API changes.